tests/dags/providers/provider_api_scripts/test_nypl.py

```diff
@@ -145,54 +107,51 @@ def test_get_metadata_missing_attrs():
     mods["physicalDescription"].pop("note")
     # Remove data from expected values too
     expected_metadata = _get_resource_json("metadata.json")
-    for attr in ["date_issued", "publisher", "description"]:
+    for attr in ["date_issued", "publisher", "physical_description"]:
         expected_metadata.pop(attr)
 
-    actual_metadata = np._get_metadata(mods)
+    actual_metadata = nypl._get_metadata(mods)
 
     assert actual_metadata == expected_metadata
 
 
-def test_handle_results_success():
+def test_get_record_data_success():
     search_response = _get_resource_json("response_search_success.json")
-    result = search_response.get("nyplAPI").get("response").get("result")
-
+    result = search_response["nyplAPI"]["response"]["result"][0]
     item_response = _get_resource_json("response_itemdetails_success.json")
-    r = requests.Response()
-    r.status_code = 200
-    r.json = MagicMock(return_value=item_response)
-    with patch.object(np.delay_request, "get", return_value=r) as mock_request:
-        with patch.object(np.image_store, "add_item") as mock_item:
-            np._handle_results(result)
-
-    assert mock_item.call_count == 7
-    assert mock_request.call_count == 1
 
-
-def test_handle_results_failure():
+    with patch.object(nypl, "get_response_json", return_value=item_response):
+        images = nypl.get_record_data(result)
+    assert len(images) == 7
+    expected_image = {
+        "category": None,
+        "creator": "Hillman, Barbara",
+        "filetype": "jpeg",
+        "foreign_identifier": "56738462",
+        "foreign_landing_url": "http://digitalcollections.nypl.org/items/0cabe3d0-3d50-0134-a8e0-00505686a51c",
+        "image_url": "http://images.nypl.org/index.php?id=56738462&t=g&suffix=0cabe3d0-3d50-0134-a8e0-00505686a51c.001",
+        "meta_data": {
+            "date_issued": "1981",
+            "genre": "Maps",
+            "publisher": "New York Public Library, Local History and Genealogy Division",
+            "tags": "Census districts",
+            "type_of_resource": "cartographic",
+            "physical_description": "4 polyester film encapsulations, some containing 2 sheets back-to-back. "
+            "Accompanying text formatted as 1 large sheet (46 x 59 cm), in one of "
+            "the encapsulations.",
+        },
+        "title": "1900 census enumeration districts, Manhattan and Bronx",
+        "license_info": CC0,
+    }
+    for key, value in expected_image.items():
+        assert images[0][key] == value
```

Comment:
Pytest should be able to show a reasonable diff for this output, I think we can just do (unless there are more fields present in `image[0]` we don't want to check against):
```suggestion
    assert images[0] == expected_image
```